### ISSUE TITLE
HW29

### ISSUE THREAD

**Issue by MarynaBasieva** [katrinsky/katrinsky#19]
Все вірно, але є деякі дрібні недоліки, які потрібно виправити

В конструкторі класу AuthException ви повинні спочатку викликати конструктор батьківського класу Error, використовуючи super().

У рядку "You don't have access to this page" використовується апостроф, який не екранується. Це призведе до синтаксичної помилки. Вам слід екранувати апострофи в рядку або використовувати одинарні лапки навколо цього рядка.

**Comment by katrinsky** [comment]
Я наче викликала за допомогою super(), але бачу не передала йому аргумент message. Тобто має виглядати ось так? Чи буде це вірно?
class AuthException extends Error {
  constructor(code, message) {
    super(message); 
    this.code = code;
    this.message = message ? `${code}: ${message}` : `${code}`;
  }

По другій правці, в мене цей вираз стоїть у подвійних лапках (у завданні він у апострофах). Хіба в моєму варіанті потрібне додаткове екранування?

**Comment by MarynaBasieva** [comment]
1) При використані двох лапок можна не екранувати але все ж таки іноді може призвезти до помилки . Тому рекомендую екранувати. Зараз це не рахується помилкою, а як рекомендація. 


Я наче викликала за допомогою super(), але бачу не передала йому аргумент message. Тобто має виглядати ось так? Чи буде це вірно?
class AuthException extends Error {
constructor(code, message) {
super(message);
this.code = code;
this.message = message ? ${code}: ${message} : ${code};
}


2) Так саме так

**Comment by katrinsky** [comment]
Я зрозуміла, дякую. На лекції нам просто казали, що екрануються ті частини, які дублюються. Тобто одинарні лапки в одинарних або подвійні в подвійних. Тому я якось не задумуючись оголосила рядок подвійними, щоб не було помилки :)

**Comment by MarynaBasieva** [comment]
Так, вірно зробили але коли використовуєте одинарні обов'язково екрануйте

**Comment by katrinsky** [comment]
Це обов'язково! :)
